### ISSUE TITLE
Unhandled Rejection: Syntax error when trying to run Build or Develop when using ACF to REST API Wordpress plugin

### ISSUE THREAD

**Issue by michaelmccann** [gatsbyjs/gatsby#20530]
## Description
I am currently trying to convert a website from Wordpress to a headless Wordpress CMS using Gatsby. 
I am using the gatsby default starter theme as a base, and I also have the gatsby-source-wordpress plugin as well as having ACF Pro and the ACF to REST API plugin. My site uses ACF Pro extensively throughout, especially a lot of Flexible Content, Repeater fields, etc. Currently, I am just trying to get everything working before starting to convert the site, hence using a default starter theme.

When I run gatsby develop or build I get an 
`UNHANDLED REJECTION: Syntax Error: Unexpected Name 'WordPressAcf_homepage' `
error.

The gatsby-source-wordpress plugin option are as follows: 
```
resolve: `gatsby-source-wordpress`,
      options: {
        // your WordPress source
        baseUrl: `stage-www.mypossibleself.com`,
        protocol: `https`,
        // is it hosted on wordpress.com, or self-hosted?
        hostingWPCOM: false,
        // does your site use the Advanced Custom Fields Plugin?
        useACF: true,
        acfOptionPageIds: [],
        includedRoutes: [
          "**/categories",
          "**/posts",
          "**/pages",
          "**/media",
          "**/menus"
        ],
      }
```
I don't get this issue when I use a new WP install with no data - Gatsby compiles correctly. It's only when I try to pull in data from the website I am trying to convert. 

### Steps to reproduce
Set up a new Gatsby site using the Gatsby default starter theme and use the following options for the gatsby-source-wordpress plugin in gatsby-config.js: 
```
resolve: `gatsby-source-wordpress`,
      options: {
        // your WordPress source
        baseUrl: `stage-www.mypossibleself.com`,
        protocol: `https`,
        // is it hosted on wordpress.com, or self-hosted?
        hostingWPCOM: false,
        // does your site use the Advanced Custom Fields Plugin?
        useACF: true,
        acfOptionPageIds: [],
        includedRoutes: [
          "**/categories",
          "**/posts",
          "**/pages",
          "**/media",
          "**/menus"
        ],
      }
```
Run GATSBY_CONCURRENT_DOWNLOAD=10 gatsby develop. 

### Expected result

Gatsby should compile correctly with no errors. I should be able to then use GraphQL to query the site and pull in the ACF info. 

### Actual result

When I run gatsby develop or build, I get the following errors:
```
UNHANDLED REJECTION Syntax Error: Unexpected Name "WordPressAcf_homepage"

  GraphQLError: Syntax Error: Unexpected Name "WordPressAcf_homepage"
  
  - TypeMapper.js:113 TypeMapper.createType
    [mps-gatsby-test]/[graphql-compose]/lib/TypeMapper.js:113:43
  
  - ObjectTypeComposer.js:80 Function.createTemp
    [mps-gatsby-test]/[graphql-compose]/lib/ObjectTypeComposer.js:80:28
  
  - ObjectTypeComposer.js:56 Function.create
    [mps-gatsby-test]/[graphql-compose]/lib/ObjectTypeComposer.js:56:21
  
  - index.js:59 typeNames.forEach.typeName
    [mps-gatsby-test]/[gatsby]/dist/schema/infer/index.js:59:41
  
  - Array.forEach
  
  - index.js:42 addInferredTypes
    [mps-gatsby-test]/[gatsby]/dist/schema/infer/index.js:42:13
  
  - schema.js:184 updateSchemaComposer
    [mps-gatsby-test]/[gatsby]/dist/schema/schema.js:184:9
```
I cannot find any reverence to WordPressAcf_homepage anywhere in my code, or in the code for the wordpress site, so I'm a bit baffled. As I said earlier, everything works fine on a blank WP install using ACF Pro and ACF to REST API plugins. Everything works fine if I DON'T use the ACF to REST API plugin. 

### Environment

System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.10.0 - /usr/local/bin/node
    npm: 5.6.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 79.0.3945.117
    Firefox: 71.0
    Safari: 13.0.4
  npmPackages:
    gatsby: ^2.18.12 => 2.18.12
    gatsby-image: ^2.2.34 => 2.2.34
    gatsby-plugin-manifest: ^2.2.31 => 2.2.31
    gatsby-plugin-offline: ^3.0.27 => 3.0.27
    gatsby-plugin-react-helmet: ^3.1.16 => 3.1.16
    gatsby-plugin-sharp: ^2.3.5 => 2.3.5
    gatsby-source-filesystem: ^2.1.40 => 2.1.40
    gatsby-source-wordpress: ^3.1.53 => 3.1.53
    gatsby-transformer-sharp: ^2.3.7 => 2.3.7
  npmGlobalPackages:
    gatsby-cli: 2.7.57

**Comment by TylerBarnes** [comment]
Hi @michaelmccann !

Sorry to hear you're running into an issue. To help us best begin debugging the underlying cause, it is incredibly helpful if you're able to create a [minimal reproduction][reproduction]. This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

If you're up for it, we'd very much appreciate if you could [provide a minimal reproduction][reproduction] and we'll be able to take another look.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by michaelmccann** [comment]
Minimal reproduction here: https://github.com/Micanio/gatbsy-bug-repro.git
FYI, I've also tried this with the latest versions of npm and node installed, just in case that was the issue, but keep getting the same error.

**Comment by TylerBarnes** [comment]
Hi @michaelmccann , apologies for the late reply on this. I've been very busy working on a new version of `gatsby-source-wordpress` which uses WPGraphQL instead of the REST API, precisely to fix the kinds of problems you're seeing here which are inherent to using an untyped REST API as a source. I had a quick look and I believe you're having the same problem as this issue https://github.com/gatsbyjs/gatsby/issues/20981#issuecomment-583788289

I would recommend using `gatsby-source-graphql` with WPGraphQL for now, instead of using `gatsby-source-wordpress` so that you wont have this issue. There may be a way to fix this in the current version of `gatsby-source-wordpress` but I haven't been able to identify an easy way to fix it. I'm going to close this as a duplicate of that issue, but if you have any additional questions, feel free to ask here or in that other issue.
